TranslationAssistant.Business/DocumentTranslationManager.cs

```diff
@@ -269,29 +293,39 @@ private static string ConvertToXlsx(string fullPath, string targetLanguage)
         /// <param name="fullNameForDocumentToProcess">The full name for document to process.</param>
         /// <param name="sourceLanguage">The source language.</param>
         /// <param name="targetLanguage">The target langauge.</param>
-        private static void DoTranslationInternal(
+        private static int[] DoTranslationInternal(
             string fullNameForDocumentToProcess,
             string sourceLanguage,
             string targetLanguage,
             bool ignoreHidden = false)
         {
+            int sourceCharCount = 0;
+            int targetCharCount = 0;
             try
             {
                 if (fullNameForDocumentToProcess.ToLowerInvariant().EndsWith(".docx"))
                 {
-                    ProcessWordDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage, ignoreHidden);
+                    int[] intArray = ProcessWordDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage, ignoreHidden);
+                    sourceCharCount = intArray[0];
+                    targetCharCount = intArray[1];
                 }
                 else if (fullNameForDocumentToProcess.ToLowerInvariant().EndsWith(".xlsx"))
                 {
-                    ProcessExcelDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage, ignoreHidden);
+                    int[] intArray = ProcessExcelDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage, ignoreHidden);
+                    sourceCharCount = intArray[0];
+                    targetCharCount = intArray[1];
                 }
                 else if (fullNameForDocumentToProcess.ToLowerInvariant().EndsWith(".pptx"))
                 {
-                    ProcessPowerPointDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage, ignoreHidden);
+                    int[] intArray = ProcessPowerPointDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage, ignoreHidden);
+                    sourceCharCount = intArray[0];
+                    targetCharCount = intArray[1];
                 }
                 else if (fullNameForDocumentToProcess.ToLowerInvariant().EndsWith(".txt") || fullNameForDocumentToProcess.ToLowerInvariant().EndsWith(".text"))
                 {
-                    ProcessTextDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage);
+                    int[] intArray = ProcessTextDocument(fullNameForDocumentToProcess, sourceLanguage, targetLanguage);
```

Comment:
I would go with returning `DocumentTranslatorResult` all the way up. These should just be assigning a DocumentTranslatorResult variable to be returned. Array isn't a very understandable interface